node/src/blockchain/blockchain_interface/mod.rs

```diff
@@ -1,52 +1,39 @@
 // Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
 
-pub mod blockchain_interface_null;
 pub mod blockchain_interface_web3;
 pub mod data_structures;
 pub mod lower_level_interface;
 pub mod test_utils;
 
-use crate::accountant::db_access_objects::payable_dao::PayableAccount;
 use crate::accountant::scanners::mid_scan_msg_handling::payable_scanner::blockchain_agent::BlockchainAgent;
-use crate::blockchain::blockchain_bridge::PendingPayableFingerprintSeeds;
 use crate::blockchain::blockchain_interface::data_structures::errors::{
-    BlockchainAgentBuildError, BlockchainError, PayableTransactionError, ResultForReceipt,
-};
-use crate::blockchain::blockchain_interface::data_structures::{
-    ProcessedPayableFallible, RetrievedBlockchainTransactions,
+    BlockchainAgentBuildError, BlockchainError,
 };
+use crate::blockchain::blockchain_interface::data_structures::RetrievedBlockchainTransactions;
 use crate::blockchain::blockchain_interface::lower_level_interface::LowBlockchainInt;
-use crate::db_config::persistent_configuration::PersistentConfiguration;
 use crate::sub_lib::wallet::Wallet;
-use actix::Recipient;
-use web3::types::{Address, BlockNumber, H256};
+use futures::Future;
+use masq_lib::blockchains::chains::Chain;
+use web3::types::{Address, BlockNumber};
 
 pub trait BlockchainInterface {
     fn contract_address(&self) -> Address;
 
+    fn get_chain(&self) -> Chain;
+
+    fn lower_interface(&self) -> Box<dyn LowBlockchainInt>;
```

Comment:
When I'm looking at this, I'm sad, because I wish to bring up something that might be drastic too much. You can hardly know it because you were not at it, but this "lower level interface" had a more distinct purpose than in your setup. 

It was supposed to embrace methods with these similarities:

a) They should work as or almost as simple wrapping methods. They should contain straightforward implementations of code from any library we would use to deal with the blockchain remotely (RPC handling methods).

b) The main purpose of this interface was to allow use the method of mocking those RPC calls in complex tests. 
I know you brought over a different approach, engaging heavily the MockBlockchainClientServer. It was a decision which led you by a trajectory around what this lower_interface was intended for.

One big problem is that you put some methods of this main trait `BlockchainInterface` in the back, while some resisted in the front.

While it seems completely reasonable to have this main interface including convenient methods like `retrieve_transactions`, `build_blockchain_agent` methods, implying a heavier functionality (and thus requiring a bunch of code), it looks like you forgot to accompany those with also other methods of the same caliber. Such are: `send_payables_in_batch`, possibly the future `request_transaction_receipts_IN_BATCH`.   

Notice of the thing that links all those four together: They are each an encapsulated independent procedure, which may or may not run something complex to bring you back a clearly represented, highly utilizable  kind of response (often our own code structure and not something from the blockchain library). 
This is what I don't really like on your lay out, it mixes high level functions with the simple ones belonging only to the low_level_interface. 

By which I just said that despite you found an argument you can fight with against me, with your more-laborious-but-also-impressive use of the perfectly configured mock server, which you neutralize the need for mockable methods in our code, yet you probably should rearrange this main interface to include all methods of the "BIG FOUR".
It would feel right to do for me, and I'm writing this having in mind that the changes will cost some extra time making them. 
 
This is the kind of comment, which is more architectural, and I want to encourage you to consult with our architect Dan prior you get down to it.